src/metadata.rs

```diff
@@ -14,17 +14,21 @@
  * limitations under the License.
  */
 
-use std::{collections::HashMap, convert::TryFrom, sync::Arc};
-
-use crate::xds::config::core::v3::Metadata as ProtoMetadata;
+pub(crate) mod symbol;
 
 #[doc(hidden)]
 pub mod build {
     include!(concat!(env!("OUT_DIR"), "/built.rs"));
 }
 
+use std::{collections::HashMap, convert::TryFrom};
+
+use crate::xds::config::core::v3::Metadata as ProtoMetadata;
+
+pub use symbol::{Key, Reference, Symbol};
```

Comment:
Just checking if I'm missing something, but are Reference/Symbol used anywhere yet?

Reply:
No they aren't, they will be used in a PR I'll put up today that updates `ConcatencateBytes` to accept `Symbol`s instead of just raw bytes, letting you accept bytes from metadata.

Reply:
Gotcha - that makes sense. Was going to ask for docs, but if it's not actually being used yet, we can hold off on that 👍🏻